src/test/java/uk/nhs/tis/trainee/notifications/migration/ResendGoogleMailFailuresIntegrationTest.java

```diff
@@ -0,0 +1,179 @@
+ */
+
+package uk.nhs.tis.trainee.notifications.migration;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.params.provider.EnumSource.Mode.EXCLUDE;
+import static uk.nhs.tis.trainee.notifications.TestContainerConfiguration.MONGODB;
+import static uk.nhs.tis.trainee.notifications.model.MessageType.EMAIL;
+import static uk.nhs.tis.trainee.notifications.model.NotificationStatus.FAILED;
+import static uk.nhs.tis.trainee.notifications.model.NotificationType.WELCOME;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import jakarta.annotation.Nullable;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import org.bson.types.ObjectId;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.test.context.ActiveProfiles;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import uk.nhs.tis.trainee.notifications.model.History;
+import uk.nhs.tis.trainee.notifications.model.History.RecipientInfo;
+import uk.nhs.tis.trainee.notifications.model.MessageType;
+import uk.nhs.tis.trainee.notifications.model.NotificationStatus;
+
+@SpringBootTest
+@ActiveProfiles("test")
+@Testcontainers(disabledWithoutDocker = true)
+class ResendGoogleMailFailuresIntegrationTest {
+
+  private static final String LOG_MESSAGE = "Found 2 qualifying failure(s).";
+
+  private static final Logger log = (Logger) LoggerFactory.getLogger(
+      ResendGoogleMailFailures.class);
+
+  @Container
+  @ServiceConnection
+  private static final MongoDBContainer MONGODB_CONTAINER = new MongoDBContainer(MONGODB);
+
+  @SpyBean
+  private MongoTemplate mongoTemplate;
+
+  private ResendGoogleMailFailures migrator;
+  private List<ILoggingEvent> logsList;
+
+  @BeforeEach
+  void setUp() {
+    ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+    listAppender.start();
+    log.addAppender(listAppender);
+    logsList = listAppender.list;
+
+    mongoTemplate.dropCollection(History.class);
+    migrator = new ResendGoogleMailFailures(mongoTemplate, null, null, null);
+  }
+
+  @ParameterizedTest
+  @EnumSource(value = MessageType.class, mode = EXCLUDE, names = "EMAIL")
+  void shouldFindByRecipientType(MessageType type) {
+    createFailure(null, null, null, null, null);
```

Comment:
Given that `createFailure()` is part of the test class, and itself replaces nulls with values that would match the filter, is it useful to include` createFailure(null, null, etc)` as a test entry?

Reply:
I guess it tests the createFailure() logic, so ok. Inception...